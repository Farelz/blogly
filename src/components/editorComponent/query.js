import gql from 'graphql-tag';

export const createPost = gql`
    mutation createPost($title: String!, $body: String!, $postPicUrl: String, $tag: String) {
        createPost(title: $title, body: $body, postPicUrl: $postPicUrl, tag: $tag) {
            post {
                id
            }
        }
    }
`;

export default gql`
    query UserNewPostData {
        user {
            id
            fullName
            description
            gravatarUrl
        }
    }
`;
