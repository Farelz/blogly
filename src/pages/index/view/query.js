import gql from 'graphql-tag';

export default gql`
    query AllPosts($after: String, $first: Int, $last: Int, $before: String) {
        allPost(first: $first, last: $last, before: $before, after: $after, sort: timestamp_desc) {
            edges {
                node {
                    id
                    title
                    timestamp
                    postPicUrl
                    body
                    author {
                        gravatarUrl
                        fullName
                        id
                    }
                }
            }
            pageInfo {
                startCursor
                endCursor
                hasNextPage
                hasPreviousPage
            }
            totalCount
        }
    }
`;
