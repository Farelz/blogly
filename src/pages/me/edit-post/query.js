import gql from 'graphql-tag';

export default gql`
    query PostToEdit($post_name: String!) {
        post(title: $post_name) {
            uuid
            title
            body
            postPicUrl
            tag {
                name
            }
        }
    }
`;
