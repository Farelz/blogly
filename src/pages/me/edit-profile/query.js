import gql from 'graphql-tag';

export const updateProfile = gql`
    mutation updateProfile($newName: String, $newDesc: String) {
        updateUserInfo(newFullName: $newName, newDescription: $newDesc) {
            user {
                id
                fullName
                description
            }
        }
    }
`;

export const updateProfilePic = gql`
    mutation updateProfile($newPic: String!) {
        updateUserProfilePic(newPic: $newPic) {
            user {
                id
            }
        }
    }
`;

export default gql`
    query profileUpdateData {
        user {
            id
            uuid
            fullName
            description
            gravatarUrl
        }
    }
`;
