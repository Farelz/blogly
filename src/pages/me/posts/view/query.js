import gql from 'graphql-tag';

export const deleteMutation = gql`
    mutation deletePost($postId: Int!) {
        deletePost(postId: $postId) {
            post {
                id
            }
        }
    }
`;

export default gql`
    query UserPosts($first: Int, $after: String) {
        user {
            id
            posts(first: $first, after: $after) {
                edges {
                    node {
                        uuid
                        id
                        title
                        timestamp
                        postPicUrl
                        body
                    }
                }
                pageInfo {
                    endCursor
                    hasNextPage
                }
            }
        }
    }
`;
