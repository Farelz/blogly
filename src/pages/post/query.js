import gql from 'graphql-tag';

export default gql`
    query Post($post_name: String) {
        post(title: $post_name) {
            uuid
            title
            postPicUrl
            body
            timestamp
            author {
                id
                gravatarUrl
                fullName
                description
            }
            tag {
                name
            }
            claps {
                totalCount
            }
            comments {
                edges {
                    node {
                        uuid
                        id
                        body
                        timestamp
                        author {
                            id
                            fullName
                            gravatarUrl
                        }
                        replies {
                            edges {
                                node {
                                    id
                                    body
                                    timestamp
                                    author {
                                        id
                                        fullName
                                        gravatarUrl
                                    }
                                }
                            }
                        }
                    }
                }
            }
            nextTitle
            prevTitle
        }
    }
`;
