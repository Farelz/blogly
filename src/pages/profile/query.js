import gql from 'graphql-tag';

export default gql`
    query PublicProfile($user_name: String) {
        publicUser(name: $user_name) {
            id
            fullName
            description
            memberSince
            gravatarUrl
            posts(first: 5) {
                edges {
                    node {
                        id
                        title
                        timestamp
                        body
                        postPicUrl
                    }
                }
            }
        }
    }
`;
