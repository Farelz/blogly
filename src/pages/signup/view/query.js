import gql from 'graphql-tag';

export default gql`
    mutation createUser(
        $fullName: String!
        $email: String!
        $password: String!
        $username: String!
    ) {
        createUser(fullName: $fullName, email: $email, password: $password, username: $username) {
            token
        }
    }
`;
