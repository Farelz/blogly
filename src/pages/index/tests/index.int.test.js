import React from 'react';
import wait from 'waait';

import { MockedProvider } from 'react-apollo/test-utils';
import { render } from 'react-testing-library';

import mocker from '../../../../mocks/materializeMock';

import Index from '../index';
import query from '../view/query';

import result from './query_result';

describe('Homepage integration tests', () => {
    const mocks = {
        request: {
            query: query,
            variables: {
                first: 6
            }
        },
        result
    };

    const getTitles = (nodes) => {
        let titles = [];
        nodes.forEach((node) => {
            titles = titles.concat(node.text);
        });

        return titles;
    };

    const page = async (mock) => {
        mocker('Sidenav', jest.fn());
        const page = render(
            <MockedProvider mocks={[mock]} addTypename={false}>
                <Index loggedIn={false} />
            </MockedProvider>
        );

        await wait(0);

        return page;
    };

    it('render all titles', async () => {
        const { container } = await page(mocks);
        const postNode = container.querySelectorAll('a.post-card__title--link');

        const postTitles = getTitles(postNode);

        expect(postNode).toHaveLength(6);
        expect(postTitles).toEqual([
            'Bio: Andy Bernard',
            'Test Markdown styles',
            'Refactor bug',
            'Hello New World',
            'Hello World',
            'Quam non nam?'
        ]);
        expect(container).toMatchSnapshot();
    });

    test('error state', async () => {
        const mock = {
            ...mocks,
            error: new Error('aw shucks')
        };

        const { container } = await page(mock);
        const errorNode = container.querySelector('div.error p');

        expect(errorNode.textContent).toContain('Error fetching posts');
    });
});
