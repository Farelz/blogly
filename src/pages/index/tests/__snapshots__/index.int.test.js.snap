// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage integration tests render all titles 1`] = `
<div>
  <div
    class="navbar-wrapper"
  >
    <nav
      class="topbar nav-extended"
    >
      <div
        class="nav-wrapper"
      >
        <a
          class="sidenav-trigger hide-on-med-and-up"
          data-target="mobile-topbar"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align: middle;"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m5 10h30v3.4h-30v-3.4z m0 11.6v-3.2h30v3.2h-30z m0 8.4v-3.4h30v3.4h-30z"
              />
            </g>
          </svg>
        </a>
        <a
          class="brand-logo center"
          href="/"
        >
          <span
            class="blog-title"
          >
            ReactPress
          </span>
        </a>
        <ul
          class="right hide-on-small-only"
        >
          <li>
            <a
              class="sign-in"
              href="/login"
            >
              Sign in
            </a>
          </li>
          <li>
            <a
              class="sign-up btn"
              href="/signup"
            >
              Get started
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <ul
      class="sidenav hide-on-med-and-up"
      id="mobile-topbar"
    >
      <li>
        <div
          class="user-view"
        >
          <a>
            <img
              class="circle defImg"
              src="/static/default-pic.png"
            />
          </a>
          <span
            class="email"
          >
            Hello, 
            Guest
          </span>
        </div>
      </li>
      <li>
        <a
          class="hide-on-med-and-up"
          href="/login"
        >
          Sign In
        </a>
      </li>
      <div
        class="divider"
      />
      <li>
        <a
          class="hide-on-med-and-up"
          href="/signup"
        >
          Get Started
        </a>
      </li>
    </ul>
  </div>
  <div
    class="home-body section container"
  >
    <div
      class="row"
    >
      <div
        class="col l6 s12"
      >
        <div>
          <div
            class="post-card card horizontal"
          >
            <div
              class="card-stacked"
            >
              <div
                class="card-content"
              >
                <h1
                  class="post-card__title"
                >
                  <a
                    class="post-card__title--link"
                    href="/p/bio%3A%20andy%20bernard"
                  >
                    Bio: Andy Bernard
                  </a>
                </h1>
                <div
                  class="post-card__text"
                >
                  Test response
                </div>
              </div>
              <div
                class="post-card__meta card-action"
              >
                <div
                  class="post-card__meta__author"
                >
                  <div
                    class="post-card__meta__author-image"
                  >
                    <img
                      class="circle"
                      src="/static/default-pic.png"
                    />
                  </div>
                  <div
                    class="post-card__meta__info"
                  >
                    <a
                      href="/profile/marvin%20kome"
                      title="Marvin Kome"
                    >
                      <span
                        class="post-card__meta__name"
                      >
                        Marvin Kome
                      </span>
                    </a>
                    <span
                      class="post-card__meta__date"
                    >
                      Jul 21
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col l6 s12"
      >
        <div>
          <div
            class="post-card card horizontal"
          >
            <div
              class="card-stacked"
            >
              <div
                class="card-content"
              >
                <h1
                  class="post-card__title"
                >
                  <a
                    class="post-card__title--link"
                    href="/p/test%20markdown%20styles"
                  >
                    Test Markdown styles
                  </a>
                </h1>
                <div
                  class="post-card__text"
                >
                  Test response
                </div>
              </div>
              <div
                class="post-card__meta card-action"
              >
                <div
                  class="post-card__meta__author"
                >
                  <div
                    class="post-card__meta__author-image"
                  >
                    <img
                      class="circle"
                      src="/static/default-pic.png"
                    />
                  </div>
                  <div
                    class="post-card__meta__info"
                  >
                    <a
                      href="/profile/marvin%20kome"
                      title="Marvin Kome"
                    >
                      <span
                        class="post-card__meta__name"
                      >
                        Marvin Kome
                      </span>
                    </a>
                    <span
                      class="post-card__meta__date"
                    >
                      Jul 21
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col l6 s12"
      >
        <div>
          <div
            class="post-card card horizontal"
          >
            <div
              class="card-stacked"
            >
              <div
                class="card-content"
              >
                <h1
                  class="post-card__title"
                >
                  <a
                    class="post-card__title--link"
                    href="/p/refactor%20bug"
                  >
                    Refactor bug
                  </a>
                </h1>
                <div
                  class="post-card__text"
                >
                  Trying to refactor a bug

                </div>
              </div>
              <div
                class="post-card__meta card-action"
              >
                <div
                  class="post-card__meta__author"
                >
                  <div
                    class="post-card__meta__author-image"
                  >
                    <img
                      class="circle"
                      src="/static/default-pic.png"
                    />
                  </div>
                  <div
                    class="post-card__meta__info"
                  >
                    <a
                      href="/profile/marvin%20kome"
                      title="Marvin Kome"
                    >
                      <span
                        class="post-card__meta__name"
                      >
                        Marvin Kome
                      </span>
                    </a>
                    <span
                      class="post-card__meta__date"
                    >
                      Jul 21
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col l6 s12"
      >
        <div>
          <div
            class="post-card card horizontal"
          >
            <div
              class="card-stacked"
            >
              <div
                class="card-content"
              >
                <h1
                  class="post-card__title"
                >
                  <a
                    class="post-card__title--link"
                    href="/p/hello%20new%20world"
                  >
                    Hello New World
                  </a>
                </h1>
                <div
                  class="post-card__text"
                >
                  Test response
                </div>
              </div>
              <div
                class="post-card__meta card-action"
              >
                <div
                  class="post-card__meta__author"
                >
                  <div
                    class="post-card__meta__author-image"
                  >
                    <img
                      class="circle"
                      src="/static/default-pic.png"
                    />
                  </div>
                  <div
                    class="post-card__meta__info"
                  >
                    <a
                      href="/profile/marvin%20kome"
                      title="Marvin Kome"
                    >
                      <span
                        class="post-card__meta__name"
                      >
                        Marvin Kome
                      </span>
                    </a>
                    <span
                      class="post-card__meta__date"
                    >
                      Jul 19
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col l6 s12"
      >
        <div>
          <div
            class="post-card card horizontal"
          >
            <div
              class="card-stacked"
            >
              <div
                class="card-content"
              >
                <h1
                  class="post-card__title"
                >
                  <a
                    class="post-card__title--link"
                    href="/p/hello%20world"
                  >
                    Hello World
                  </a>
                </h1>
                <div
                  class="post-card__text"
                >
                  Hey

                </div>
              </div>
              <div
                class="post-card__meta card-action"
              >
                <div
                  class="post-card__meta__author"
                >
                  <div
                    class="post-card__meta__author-image"
                  >
                    <img
                      class="circle"
                      src="/static/default-pic.png"
                    />
                  </div>
                  <div
                    class="post-card__meta__info"
                  >
                    <a
                      href="/profile/marvin%20kome"
                      title="Marvin Kome"
                    >
                      <span
                        class="post-card__meta__name"
                      >
                        Marvin Kome
                      </span>
                    </a>
                    <span
                      class="post-card__meta__date"
                    >
                      Jul 19
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col l6 s12"
      >
        <div>
          <div
            class="post-card card horizontal"
          >
            <div
              class="card-stacked"
            >
              <div
                class="card-content"
              >
                <h1
                  class="post-card__title"
                >
                  <a
                    class="post-card__title--link"
                    href="/p/quam%20non%20nam%3F"
                  >
                    Quam non nam?
                  </a>
                </h1>
                <div
                  class="post-card__text"
                >
                  Integer ac neque. Nunc nisl.
                </div>
              </div>
              <div
                class="post-card__meta card-action"
              >
                <div
                  class="post-card__meta__author"
                >
                  <div
                    class="post-card__meta__author-image"
                  >
                    <img
                      class="circle"
                      src="/static/default-pic.png"
                    />
                  </div>
                  <div
                    class="post-card__meta__info"
                  >
                    <a
                      href="/profile/janet%20cole"
                      title="Janet Cole"
                    >
                      <span
                        class="post-card__meta__name"
                      >
                        Janet Cole
                      </span>
                    </a>
                    <span
                      class="post-card__meta__date"
                    >
                      Jul 08
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row pagination"
    />
  </div>
</div>
`;
